src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -21,27 +21,40 @@ namespace Voron.Impl
     public unsafe class LowLevelTransaction : IDisposable
     {
         private const int PagesTakenByHeader = 1;
+
         public readonly AbstractPager DataPager;
         private readonly StorageEnvironment _env;
         private readonly long _id;
         private readonly ByteStringContext _allocator;
         private readonly bool _disposeAllocator;
-        private Tree _root;
 
-        internal FixedSizeTree _freeSpaceTree;
+        private Tree _root;
+        public Tree RootObjects => _root;        
 
-        public bool FlushedToJournal;
-        public Tree RootObjects => _root;
+        public bool FlushedToJournal;        
 
         private readonly WriteAheadJournal _journal;
+        internal readonly List<JournalSnapshot> JournalSnapshots = new List<JournalSnapshot>();
 
+        private static ObjectPool<Dictionary<long, PageFromScratchBuffer>, DictionaryResetBehavior<long, PageFromScratchBuffer>> ScratchPagesTablePool = new ObjectPool<Dictionary<long, PageFromScratchBuffer>, DictionaryResetBehavior<long, PageFromScratchBuffer>>(() => new Dictionary<long, PageFromScratchBuffer>(NumericEqualityComparer.Instance), 30);
```

Comment:
Thats actually a good idea. In that way each pool will behave closely to its typical use. And we can also react to low memory conditions